packaging/installer/netdata-updater.sh

```diff
@@ -151,11 +151,26 @@ issystemd() {
 
 # shellcheck disable=SC2009
 running_under_anacron() {
-    ppid="$(ps -o pid=,ppid= | grep -e "^ $$" | xargs | cut -f 2 -d ' ')"
-    ps -o pid=,command= | grep -e "^ ${ppid}" | grep -q anacron && return 0
+    if [ "$(uname -s)" = "Linux" ] && [ -d "/proc/$$" ]; then
+        ppid="$(cut -f 4 -d ' ' "/proc/$$/stat")"
+        if [ -n "${ppid}" ] && [ -d "/proc/${ppid}" ]; then
+            grep -q anacron "/proc/${ppid}/comm" && return 0
 
-    ppid2="$(ps -o pid=,ppid= | grep -e "^ ${ppid}" | xargs | cut -f 2 -d ' ')"
-    ps -o pid=,command= | grep -e "^ ${ppid2}" | grep -q anacron && return 0
+            ppid2="$(cut -f 4 -d ' ' "/proc/${ppid}/stat")"
+
+            if [ -n "${ppid2}" ] && [ -d "/proc/${ppid2}" ]; then
+                grep -q anacron "/proc/${ppid2}/comm" && return 0
+            fi
+        fi
+    else
+        ppid="$(ps -o pid=,ppid= | grep -e "^ *$$" | xargs | cut -f 2 -d ' ')"
+        if [ -n "${ppid}" ]; then
+            ps -o pid=,command= | grep -e "^ ${ppid}" | grep -q anacron && return 0
```

Comment:
What is `command=`? Where is it supposed to work?

Reply:
It prints the full command line, it’s specified by POSIX, and is supported by Busybox `ps`, all versions of procps (including what’s shipped on CentOS 7) that I tested, and the implementation of `ps` on FreeBSD (both 13 and 14). I’ve not tested on macOS, but it should still work there as the utilities are still largely POSIX compliant and there’s no good reason for them to have changed this.

That said, there’s a different issue here in the command line option formatting that breaks it on at least some implementations.

Reply:
Like I said, the option formatting was broken. It should be fixed in the latest commit.